Utils/Dataflow/pyDKB/dataflow/stage/processors.py

```diff
@@ -11,61 +11,65 @@
+
+    def __init__(self):
+        super(JSONProcessorStage, self).__init__()
+        self._set_input_message_class(messageType.JSON)
+        self._set_output_message_class(messageType.JSON)
+
+    def file_input(self, fd):
+        """Override AbstractProcessorStage.file_input"""
+        try:
+            data = json.load(fd)
+            if type(data) == dict:
+                data = [data]
+            for m in data:
+                yield self.parseMessage(m)
+        except ValueError, err:
+            sys.stderr.write("(WARN) failed to read input file %s as %s: %s.\n"
+                       % (fd.name, self.input_message_class().typeName(), err))
+            yield None
 
 class TTLProcessorStage(AbstractProcessorStage):
-  """
-  Input message: TTL
-  Output message: TTL
-  """
-
-  def __init__(self):
-    super(JSONProcessorStage, self).__init__()
-    self._set_input_message_class(messageType.JSON)
-    self._set_output_message_class(messageType.JSON)
-
-  # Override
-  def output(self, message):
-    pass
-
-class JSON2TTLProcessorStage(JSONProcessorStage,TTLProcessorStage):
-  """
-  Input message: JSON
-  Output message: TTL
-  """
-
-  def __init__(self):
-    # Get __init__ method of the last but one ancestor
-    super(JSON2TTLProcessorStage.__mro__[-3], self).__init__()
-    self._set_input_message_class(messageType.JSON)
-    self._set_output_message_class(messageType.TTL)
-
-  def input(self):
-    """Override: Falls back to JSONProcessorStage.input"""
-    # Pick the method of the first parent
-    return super(JSON2TTLProcessorStage,self).input()
-
-  def output(self,message):
-    """Override: Falls back to TTLProcessorStage.output"""
-    # Skip first parent, pick the second
-    super(JSONProcessorStage,self).output(message)
+    """ TTL2TTL Processor Stage
+
+    Input message: TTL
+    Output message: TTL
+    """
+
+    def __init__(self):
+        super(JSONProcessorStage, self).__init__()
+        self._set_input_message_class(messageType.JSON)
+        self._set_output_message_class(messageType.JSON)
+
+    # Override
+    def output(self, message):
+        super(TTLProcessorStage, self).output(message)
+
+class JSON2TTLProcessorStage(JSONProcessorStage, TTLProcessorStage):
+    """ JSON2TTL Procssor Stage
+
+    Input message: JSON
+    Output message: TTL
+    """
+
+    def __init__(self):
+        # Skip direct ancestors (self, JSON, TTL)
+        super(JSON2TTLProcessorStage.__mro__[2], self).__init__()
+        self._set_input_message_class(messageType.JSON)
+        self._set_output_message_class(messageType.TTL)
+
+    def input(self):
+        """Override: Falls back to JSONProcessorStage.input"""
+        # Pick the method of the first parent
+        return super(JSON2TTLProcessorStage, self).input()
+
+    def output(self, message):
+        """Override: Falls back to TTLProcessorStage.output"""
+        # Skip first parent, pick the second
+        super(JSONProcessorStage, self).output(message)
```

Comment:
Python\`s super() construction takes next ancestor after specified, so here we skip JSONProcessor\`s `output()` and take the one of the next ancestor (namely, TTLProcessor).